weave-js/src/components/PagePanelComponents/Home/Browse3/pages/common/EmptyContent.tsx

```diff
@@ -1,14 +1,66 @@
 import {Box} from '@mui/material';
 import React, {useState} from 'react';
+import {useParams} from 'react-router-dom';
+import {toast} from 'react-toastify';
 
 import {TargetBlank} from '../../../../../../common/util/links';
 import {Button} from '../../../../../Button';
+import {useWeaveflowCurrentRouteContext} from '../../context';
 import {CreateDatasetDrawer} from '../../datasets/CreateDatasetDrawer';
+import {createNewDataset} from '../../datasets/datasetOperations';
+import {DatasetPublishToast} from '../../datasets/DatasetPublishToast';
+import {useWFHooks} from '../../pages/wfReactInterface/context';
 import {EmptyProps} from './Empty';
 import {Link} from './Links';
 
 const NewDatasetButton: React.FC = () => {
   const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const [isCreatingDataset, setIsCreatingDataset] = useState(false);
+  const {entity, project} = useParams<{entity: string; project: string}>();
+  const router = useWeaveflowCurrentRouteContext();
+  const {useObjCreate, useTableCreate} = useWFHooks();
+
+  // Get the create hooks
+  const tableCreate = useTableCreate();
+  const objCreate = useObjCreate();
+
+  const handleSaveDataset = async (name: string, rows: any[]) => {
```

Comment:
I realize i'm being greedy but can we DRY this?

Reply:
and use it in the other page too?

Reply:
yes but as a very fast follow because I want to merge and deploy the fix asap